sycl/plugins/opencl/pi_opencl.cpp

```diff
@@ -257,7 +257,8 @@ pi_result piextDeviceSelectBinary(pi_device device, pi_device_binary *images,
   // Find the appropriate device image, fallback to spirv if not found
   pi_uint32 fallback = invalid_ind;
   for (pi_uint32 i = 0; i < num_images; ++i) {
-    if (strcmp(images[i]->DeviceTargetSpec, image_target) == 0) {
+    if (strncmp(images[i]->DeviceTargetSpec, image_target,
+                strlen(image_target)) == 0) {
```

Comment:
Add a comment about why you need this change.
I do not understand why.
I guess we can remove line 248.